### PR TITLE
Add CI script to build and test ACCP

### PR DESCRIPTION
**Pull request by sp717**
### Description of changes: 

Add a CI script for ACCP to build and test against new AWS-LC changes. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and the ISC license.

### CODE REVIEW

File: tests/ci/integration/run_accp_integration.sh [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+source tests/ci/common_posix_setup.sh
+
+# Assumes script is executed from the root of aws-lc directory
+SCRATCH_FOLDER=${SRC_ROOT}/"scratch"
+ACCP_SRC="${SCRATCH_FOLDER}/amazon-corretto-crypto-provider"
+
+function build_and_test_accp() {
+  pushd "${ACCP_SRC}"
+  export JAVA_HOME=$(readlink -f /usr/bin/javac | sed "s:/bin/javac::")
```
Comment (WillChilds-Klein) [inline tests/ci/integration/run_accp_integration.sh:14]: you can probably drop the `| sed` bit here

```
$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 22.04.2 LTS
Release:        22.04
Codename:       jammy

$ readlink -f $(which java)
/usr/lib/jvm/java-11-openjdk-amd64/bin/java

$ readlink -f $(which java) | sed "s:/bin/javac::"
/usr/lib/jvm/java-11-openjdk-amd64/bin/java
```

**Review by WillChilds-Klein — DISMISSED** [review]

File: tests/ci/integration/run_accp_integration.sh [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  # Assign the JDK version we're testing as the system's default JDK and
+  # assign JAVA_HOME variable to the path. Otherwise, Ubuntu will
+  # default to the newest version of Java on the system.
```
Comment (samuel40791765) [inline tests/ci/integration/run_accp_integration.sh:17]: Nit: Seems like we are defaulting to the newest JDK version on the runner here. No issue with that, but it does make this comment obsolete.

**Review by WillChilds-Klein — APPROVED** [review]

File: tests/ci/integration/run_accp_integration.sh (hunk 2 above)
Comment (sp717) [reply tests/ci/integration/run_accp_integration.sh]: Updated!

**Comment by andrewhop** [comment]
Related to https://github.com/corretto/amazon-corretto-crypto-provider/pull/390

**Review by andrewhop — APPROVED** [review]

**Review by lilpoozie2005 — APPROVED** [review]

**Pull request merged**